components/MapViewScreen/index.tsx

```diff
@@ -51,20 +51,63 @@ export default function MapViewScreen({
     projectSize: { min: 0, max: 0 },
     location: [],
   });
+  const [filteredProjectsFromDropdowns, setFilteredProjectsFromDropdowns] =
+    useState<Project[]>(projects);
 
-  const handleFilterChange = (filter: FilterType) => {
-    console.log(filter);
+  const [filteredProjectsFromSearch, setFilteredProjectsFromSearch] =
+    useState<Project[]>(projects);
+
+  // clear filters
+  const clearFilters = () => {
+    setSelectedFilters({
+      status: [],
+      technology: [],
+      projectSize: { min: 0, max: 0 },
+      location: [],
+    });
+    setFilteredProjects(projects);
+  };
+
+  // show projects based on selected filters
+  const handleFilterButtonClick = () => {
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    const { status, technology, projectSize, location } = selectedFilters;
```

Comment:
Ohh I see, that's a good idea! I made the filters under the impression that they wouldn't need to be changed, but I agree that making the logic more flexible would be better in the long run. 

The way filtering works currently is that I have a pre-defined object of possible filters (`selectedFilters`) that the projects can filter through within the `handleFilterButtonClick` method. `selectedFilters` gets updated every time a new filter is applied or un-applied. Is there anything specifically you would want me clarify?

Reply:
Ohh I see, I think I'm doing that? The `technology` variable in line 80 is just a list that holds a bunch of strings (or is empty). Then line 82 filters all the projects by checking if its `renewable_energy_technology` attribute is within this `technology` list, without caring about specific values. Is this what you mean about not hardcoding the attributes?

Reply:
I guess, given that this PR is only for the technology filter, your implementation suffices. I was just curious if there was a way to iterate over the `selectedFilters` object so we could easily implement `status`, `projectSize`, and `location` filters as well. However doing so in a way where each filter is it's own function/reducer that, when applied to a given row, can immediately determine whether it belongs in the output. Does that make sense? It might not be necessary for this implementation, but it's one possible approach.

Reply:
Ohh I see now, you're referring to the general filters. I think we can definitely do that and it makes sense to do so- this probably will involve some mapping of the filters to the project attributes in question. And then the method can just do an agnostic loop over that mapping to filter the projects. Thanks Ronnie! I'll adjust my PR for this.